File: 3152-Special-Array-II.cpp
class Solution {
public:
    vector<bool> isArraySpecial(vector<int>& nums, vector<vector<int>>& queries) {
        vector<int> arr;
        vector<bool> rs;
        arr.push_back(0);
        for(int i=1;i<nums.size();i++){
            int val=nums[i]&1;
            int val2=nums[i-1]&1;
            if(val!=val2){
                arr.push_back(0);
            }
            else
            arr.push_back(1);
        
        }
        for(int i=1;i<nums.size() ;i++){
            arr[i]=arr[i-1]+arr[i];
        }
        for(int i=0;i<queries.size();i++){
            int v=arr[queries[i][0]]-arr[queries[i][1]];
            if(v!=0){
                rs.push_back(false);
            }
            else{
                rs.push_back(true);
            }
        }
        return rs;
        

    }
};

File: 3152-Special-Array-II.java
class Solution {
    public boolean[] isArraySpecial(int[] nums, int[][] queries) {
        // int arr[] = new int[nums.length];
        boolean res[] = new boolean[queries.length];
        int sum[] = new int[nums.length];
        for(int i=1;i<nums.length;i++){
            int e1 = nums[i-1]&1;
            int e2 = nums[i]&1;
            if(e1==e2){
                sum[i] = sum[i-1]+1;
            }
            else{
                sum[i] = sum[i-1];
            }
        }
        for(int i=0;i<queries.length;i++){
            int start = queries[i][0];
            int end = queries[i][1];
            if(sum[start]==sum[end]){
                res[i] = true;
            }
            else{
                res[i] = false;
            }
        }
        return res;
    }
}